### DISCUSSION TITLE
[E-Hentai] [error] Can't call method "parent" on an undefined value

### DISCUSSION THREAD

**Discussion by Mozi71** [Difegue/LANraragi#1147] [Q&A] [answered]
**LRR Version and OS**
Docker

**Bug Details**
[E-Hentai] [error] Can't call method "parent" on an undefined value

**Matching Logs**

[2024-12-22 12:43:40] [E-Hentai] [info] Reverse Image Search Enabled, trying now.
[2024-12-22 12:43:40] [E-Hentai] [error] Can't call method "parent" on an undefined value at /root/lanraragi/script/../lib/LANraragi/Plugin/Metadata/EHentai.pm line 203.

**Screenshots**
![image](https://github.com/user-attachments/assets/9d658926-bf86-40b1-bdd3-71a315cb9fbf)

**Comment by yuriko-aya** [comment, marked as answer]
It's likely your igneous cookie is incorrect or expired.
I had the same problem and it's fixed after change the igneous cookie with the new one.

**Comment by e2fsprogs** [comment]
I am experiencing the same issue. I re-logged into E-Hentai and reissued the igneous cookie, but the issue persists.

What I have tried:

- Re-logging into E-Hentai and reissuing the igneous cookie. The issue persists.
- Enabling/disabling thumbnail search. The issue persists.
- Other galleries work fine without any problems.

It seems to be repeatedly occurring with a specific gallery (e.g., /g/3180332/ca112c4f23/).
For testing purposes, I used the original archive download file.

**Reply by e2fsprogs** [reply]
After further investigation, it seems to be an issue related to the E-Hentai search engine indexing. Newly added galleries do not appear in search results for a certain period of time (exact duration unknown) but become searchable after some time has passed. This is not an issue with Lanraragi.
However, it would be beneficial to have graceful handling for cases where no search results are found.

**Comment by IAmCaligvla** [comment]
I'm also experiencing this issue, changing the igneous cookie or leaving it blank does nothing. I doubt it's EH hiding results as this is happening to years old galleries as well as new ones.
